### PR TITLE
Remove ability to set Google account as server

### LINKED ISSUE #5400: Remove ability to set Google account as server
- Remove all UI around creating Google Drive/Sheets project
- Show error when configuring from QR code with Google Drive/Sheets account
- When deleting Google Drive/Sheets project, warn that it can’t be created again

### PR DESCRIPTION
**Pull request by grzesiek2010**
Closes #5400 

#### What has been done to verify that this works as intended?
I've added automated tests and tested the app manually too.

#### Why is this the best possible solution? Were any other approaches considered?
I don't think there is anything to discuss here.

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?
This should block the possibility to create new google drive projects as described in the issue so this should be tested. Please also verify that scanning QR codes work as expected because that's the area of risk.

#### Do we need any specific form for testing your changes? If so, please attach one.
No.

#### Does this change require updates to documentation? If so, please file an issue [here]( https://github.com/getodk/docs/issues/new) and include the link below.
No.

#### Before submitting this PR, please make sure you have:
- [x] run `./gradlew checkAll` and confirmed all checks still pass OR confirm CircleCI build passes and run `./gradlew connectedDebugAndroidTest` locally.
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/getodk/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [x] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/getodk/collect/blob/master/docs/CODE-GUIDELINES.md#ui-components-style-guidelines)

### CODE REVIEW

File: collect_app/src/androidTest/java/org/odk/collect/android/support/CollectHelpers.kt [hunk 1]
```diff
@@ -16,4 +20,21 @@ object CollectHelpers {
         application.component = testComponent
         return testComponent
     }
+
+    @JvmStatic
+    fun addGDProject(gdProject: Project.Saved, accountName: String, testDependencies: TestDependencies) {
```
Comment (seadowg) [inline collect_app/src/androidTest/java/org/odk/collect/android/support/CollectHelpers.kt]: This should probably take a `Project.New` so we don't end up with any problems caused by specifying IDs and not letting the repo generate them itself.

File: collect_app/src/main/res/xml/server_preferences.xml [hunk 2]
```diff
@@ -8,5 +8,6 @@
         android:entryValues="@array/protocol_entry_values"
         android:key="protocol"
         android:title="@string/type"
+        android:enabled="false"
```
Comment (seadowg) [inline collect_app/src/main/res/xml/server_preferences.xml:11]: I was thinking that we should only disable this for ODK server projects, but I realize that we don't want people to accidentally switch their project from Google Drive to ODK and then not be able to switch back.

**Review by seadowg — CHANGES_REQUESTED** [review]

File: collect_app/src/androidTest/java/org/odk/collect/android/support/CollectHelpers.kt (hunk 1 above)
Comment (grzesiek2010) [reply collect_app/src/androidTest/java/org/odk/collect/android/support/CollectHelpers.kt]: Right, fixed.

**Comment by seadowg** [comment]
I'm going to go ahead and merge this and QA can take a look as it feels low risk to me. It'd be a good idea for @getodk/testers to take a look at the changes on `master` (or `next` depending on when they get round to it).

**Review by seadowg — APPROVED** [review]

**Comment by srujner** [comment]
**Tested with Success!**

Verified on device with Android 12 and 13

Verified cases:

- Regression checks and Exploratory testing on Server settings and user settings; 
- Checked #5400;
- Adding project manually or through QR code; 
- Switching between projects, adding duplicated projects; 
- Invalid/Inverted QR code;
- Screen rotation, minimize app.

**Comment by dbemke** [comment]
**Tested with Success!**

Verified on device with Android 10 and 8.1

**Pull request merged**